Shared/ViewModels/VideoPlayerManager.swift

```diff
@@ -21,6 +21,24 @@ import VLCUI
 
 class VideoPlayerManager: ViewModel {
 
+    override init() {
```

Comment:
I think I would rather have this be on the actual `NativeVideoPlayer`/`VideoPlayer` views like you do with the app instead of more notifications, the `VideoPlayerManager` objects are already there.

Reply:
I tried but I have an error that I failed to fix 😕 
<img width="1599" alt="Capture d’écran 2023-08-30 à 20 52 05" src="https://github.com/jellyfin/Swiftfin/assets/11562279/5b4b6d0a-0c37-4ae0-b946-981a330bb92f">

Reply:
Not like that, as that is also invalid Swift. Add the `Environment` and `onChange`, like you do in `SwiftfinApp`, to the video players.

Reply:
I tried moving the `.onChange(of: scenePhase)` but unfortunately the onChange block is not triggered anymore. Don't know if it's bug or if this onChange only works at the App level... 😕

Reply:
After some internet research it seems that the `.onChange(of: scenePhase)` is buggy and does not always trigger if using with a `View`.
Is it a problem to leave this stuff at the `App` level?

Reply:
This wasn't working for me on the iOS 16 simulator but now since the new iOS/Xcode RCs are out it works in the iOS 17 simulator and on device. I can only see a few things online about the phase not working and those are related to macOS or the simulator which isn't reliable.

I recommend implementing a variation of [this solution:](https://stackoverflow.com/a/62850923)

```swift

struct ScenePhaseChangeModifier: ViewModifier {
    
    @Environment(\.scenePhase)
    private var scenePhase
    
    let phase: ScenePhase
    let action: () -> Void
    
    func body(content: Content) -> some View {
        content.onChange(of: scenePhase) { newValue in
            if newValue == phase {
                action()
            }
        }
    }
}

// ViewExtensions.swift

extension View {

    func onScenePhase(_ phase: ScenePhase, _ action: @escaping () -> Void) -> some View {
        modifier(ScenePhaseChangeModifier(phase: phase, action: action))
    }
}

// VideoPlayer.swift

struct VideoPlayer: View {

// ...

    var body: some View {
		// ...
                playerView
                    .onScenePhase(.active) {
                        // play player and send play report
                    }
                    .onScenePhase(.background) {
                        // pause player and send pause report
                    }
		// ...
}

```

The reason why I just want to pause on `.background` _for now_ instead of `.inactive` is because `inactive` triggers while using the Control Center while also in the app which I don't find favorable until we implement a more comprehensible solution.

Reply:
You are right! It seems to works now on real devices (don't if it's seems the iOS 17 update or not ... ) anyway. I will continue working on it now that it works!